### ISSUE TITLE
LoweringError with object and conditionals

### ISSUE THREAD

**Issue by mpetroff** [numba/numba#3029]
I don't really know a good way to describe this issue besides just the minimal example below:

```python
import numba

@numba.jit(forceobj=True)
def test():
    a = False    # Bug still triggered if this line is removed or changed to `True`
    if a:
        raise Exception
    if a:
        b = [0]
    if a:
        b[0] = 1

test()
```

It works fine when not in object mode, if any of the three conditionals are removed, if a value is assigned to `b` outside the conditionals, or if the conditionals are reordered. The issue remains while using a tuple or NumPy array instead of the list.

I can reproduce on two machines, one with Python 3.6.4 (Anaconda) and Numba 0.38.0 and one with Python 3.5.2 and Numba 0.38.1

**Comment by stuartarchibald** [comment]
Thanks for the report. Looks like liveness analysis is wrong.

**Comment by seibert** [comment]
Closed by #3230.
